astropy/wcs/wcsapi/fitswcs.py

```diff
@@ -290,7 +290,11 @@ def world_axis_physical_types(self):
     def world_axis_units(self):
         units = []
         for unit in self.wcs.cunit:
-            if unit is None:
+            if unit is None or unit == "":
+                # some CompositeUnit objects may compare == to the empty string,
```

Comment:
I find the comment a bit confusing. Maybe
```
# Use an empty string for missing units. We include dimensionless_unscaled
# here because using VO format it would be converted to '1'.
```